README.md

```diff
@@ -340,32 +353,292 @@ The above route will just return the string "Hello World!" in the web browser wh
+```
+
+When a method is decorated with the Celery task decorator then it can be submitted to the Celery task queue.
+Will add some sleeps to demonstrate what would happen with a long running calculation. Will also tell Celery about in which step the calculation is, later we can display this step to the user.
+
+```{.python file=src/py/tasks.py}
+import time
+
+<<celery-config>>
+
+@capp.task(bind=True)
+def calculate(self, epsilon, guess):
+  if not self.request.called_directly:
+    self.update_state(state='INITIALIZING')
+  time.sleep(5)
+  from newtonraphsonpy import NewtonRaphson
+  finder = NewtonRaphson(epsilon)
+  if not self.request.called_directly:
+    self.update_state(state='FINDING')
+  time.sleep(5)
+  root = finder.find(guess)
+  return {'root': root, 'guess': guess, 'epsilon':epsilon}
+```
+
+Instead of running the calculation when the submit button is pressed. 
+We will submit the calculation task to the task queue by using the `.delay()` function.
+The submission will return a job identifier we can use later to get the status and result of the job. The web browser will redirect to a url with the job identifier in it.
+
+```{.python #py-submit}
+@app.route('/', methods=['POST'])
+def submit():
+  epsilon = float(request.form['epsilon'])
+  guess = float(request.form['guess'])
+  from tasks import calculate
+  job = calculate.delay(epsilon, guess)
+  return redirect(url_for('result', jobid=job.id))
+```
+
+The last method is to ask the Celery task queue what the status is of the job and return the result when it is succesfull.
+
+```{.python #py-result}
+@app.route('/result/<jobid>')
+def result(jobid):
+  from tasks import capp
+  job = capp.AsyncResult(jobid)
+  job.maybe_throw()
+  if job.successful():
+    result = job.get()
+    return render_template('result.html', epsilon=result['epsilon'], guess=result['guess'], root=result['root'])
+  else:
+    return job.status
+```
+
+Putting it all together
+
+```{.python file=src/py/awebapp.py}
+from flask import Flask, render_template, request, redirect, url_for
+
+app = Flask(__name__)
+
+<<py-form>>
+
+<<py-submit>>
+
+<<py-result>>
+
+if __name__ == '__main__':
+  app.run()
+```
+
+Start the web application like before with
+
+```{.awk #py-awebapp}
+PYTHONPATH=$PWD python src/py/awebapp.py
+```
+
+Tasks will be run by the Celery worker. The worker can be started with
+
+```{.awk #py-worker}
+cd src/py
+PYTHONPATH=$PWD/../.. celery -A tasks worker
+```
+
+To test web service
+1. Goto http://localhost:5000, 
+2. Submit form,
+3. Refresh result page until progress states are replaced with result.
 
 ### Web service
 
 A web application is meant for consumption by humans and web service is meant for consumption by machines or other programs.
 So instead of returning HTML pages a web service will accept and return machine readable documents like JSON documents. A web service is an application programming interface (API) based on web technologies.
 
```

Comment:
One nice addition would be just to mention about API testing. 
Maybe link to these two options?
- swagger-inspector
- postman

Reply:
Added piece about Swagger UI